### DISCUSSION TITLE
Changing policy on license from one that expires to one that doesn't

### DISCUSSION THREAD

**Discussion by lordbarrison** [keygen-sh/community#20] [Q&A] [answered]
I have 2 policies - a trial policy with a 14 day duration, and a perpetual policy with no expiration. My plan is to issue a single license that starts off as a trial, but which could later be converted to perpetual without issuing a new license.

If I create a new license with the trial policy, I can see that the license has the proper expiration. If I change the license's policy to the perpetual policy, the original expiration remains on the license.

How should this work? What am I missing?

I should clarify that I'm doing this through the admin UI and **not** through the API.

**Comment by ezekg** [comment]
Changing a policy's duration, or changing to a different policy altogether, does not retroactively change the expiry of any previously created licenses. If you transfer a license from a policy with an expiration to one without one, you will also need to set the license's expiry to `null` to remove the previously set expiry.

**Comment by ezekg** [comment, marked as answer]
Since this is a pretty common flow for trials in particular, I've introduced a new policy setting, [`transferStrategy`](https://keygen.sh/docs/api/policies/#policies-object-attrs-transferStrategy), for controlling whether or not to reset a license's expiration after its transferred to a new policy.
